app/src/main/java/com/practicum/playlistmaker/searchRecyclerPack/RecyclerSearchAdapter.kt

```diff
@@ -24,6 +26,14 @@ class RecyclerSearchAdapter(
     private val searchHistoryManager: SearchHistoryManager?
 ) : RecyclerView.Adapter<RecyclerSearchAdapter.TrackHolder>() {
 
+    companion object {
+        private const val CLICK_DEBOUNCE_DELAY = 1000L
+    }
+
+    private var isClickAllowed = true
+
+    private val handler = Handler(Looper.getMainLooper())
+
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TrackHolder {
         val itemView = LayoutInflater.from(context).inflate(R.layout.search_list_item, parent, false)
```

Comment:
```suggestion
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.search_list_item, parent, false)
```
Не нужно передавать контекст в адаптер, т.к. здесь можно использовать `parent.context`